source/common/runtime/runtime_impl.cc

```diff
@@ -42,20 +42,23 @@ void countDeprecatedFeatureUseInternal(const RuntimeStats& stats) {
   stats.deprecated_feature_seen_since_process_start_.inc();
 }
 
-#ifdef ENVOY_ENABLE_QUIC
-void refreshQuicReloadableFlags(const Snapshot::EntryMap& flag_map) {
+// TODO(12923): Document the Quiche reloadable flag setup.
+void refreshQuicheReloadableFlags(const Snapshot::EntryMap& flag_map) {
+  auto quiche_flag_map = quiche::FlagRegistry::getInstance();
+  quiche_flag_map.resetAllReloadedValue();
```

Comment:
Resetting all flags before applying the runtime feature overrides results in application code sometimes seeing the value of the flag without the runtime feature applied in cases where the old and new runtime feature value is the same.  It would be better to only reset in cases where a runtime feature is removed.  Alternatively, have this method iterate through reloadable flags and call setReloadedValue if a runtime feature override exists, else call resetReloadedValue on the flag.

Reply:
Good point.
I took the alternative route you suggested.